engine-extensions/src/main/java/io/nosqlbench/engine/extensions/computefunctions/Intersections.java

```diff
@@ -20,6 +20,32 @@
 
 public class Intersections {
 
+    /**
+     * Return a non-negative index of the first value in the sample array which is present in the reference array,
+     * OR, return a negative number. This returns array index which start at 0, not rank, which is starts at 1.
+     */
+    public static int firstMatchingIndex(long[] reference, long[] sample, int limit) {
+        Arrays.sort(reference);
```

Comment:
If this is almost always near the beginning and our limits are generally small then a brute force check of the reference list without sorting and binarysearch may be more efficient.

Reply:
Good point. We do expect it to be biased towards lower indices. Interesting optimization could happen here, but these array ops are already pretty freakin fast. Let's see if it catches on fire first.